org.eclipse.jdt.junit.core/src/org/eclipse/jdt/internal/junit/buildpath/BuildPathSupport.java

```diff
@@ -305,10 +305,10 @@ public boolean accept(File dir, String name) {
 			"org.junit", new VersionRange("[4.13.0,5.0.0)"), null, "org.junit_4.*.jar", "org.junit.source", "source-bundle/", JUnitPreferencesConstants.JUNIT4_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
 
 	private static final JUnitPluginDescription HAMCREST_CORE_PLUGIN= new JUnitPluginDescription(
-			"org.hamcrest.core", new VersionRange("[2.2.0,2.3.0)"), null, "org.hamcrest.core_2.*.jar", "org.hamcrest.core.source", "source-bundle/", JUnitPreferencesConstants.HAMCREST_CORE_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
+			"org.hamcrest.core", new VersionRange("[2.2.0,3.0.0)"), null, "org.hamcrest.core_*.jar", "org.hamcrest.core.source", "source-bundle/", JUnitPreferencesConstants.HAMCREST_CORE_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
 
 	private static final JUnitPluginDescription HAMCREST_PLUGIN= new JUnitPluginDescription(
-			"org.hamcrest", new VersionRange("[2.2.0,2.3.0)"), null, "org.hamcrest_2.*.jar", "org.hamcrest.source", "source-bundle/", JUnitPreferencesConstants.HAMCREST_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
+			"org.hamcrest", new VersionRange("[2.2.0,3.0.0)"), null, "org.hamcrest_2.*.jar", "org.hamcrest.source", "source-bundle/", JUnitPreferencesConstants.HAMCREST_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
```

Comment:
Shouldn't `2` be removed here too?

Reply:
This is just about the upper bound, the lower bound is a different/another issue (and is harder to fix), and then yes the 2.* will need attention as well.  

I will create an issue for that, because it's trickier to fix.

Reply:
No, I mean in the first entry you expect `org.hamcrest.core_*.jar` (without `2`), in the second one `org.hamcrest_2.*.jar` which looks inconsistent and strange.

Reply:
I believe Andrey means
```suggestion
			"org.hamcrest", new VersionRange("[2.2.0,3.0.0)"), null, "org.hamcrest_*.jar", "org.hamcrest.source", "source-bundle/", JUnitPreferencesConstants.HAMCREST_JAVADOC); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$
```

Reply:
Btw. there is only hamcrest 2.1 and 2.2:
https://mvnrepository.com/artifact/org.hamcrest/hamcrest

Reply:
Sorry, yes.  I started on the 1.x support but then concluded that it was more complex that desired without backing out that change.